eng/scripts/Automation-Sdk-Generate.ps1

```diff
@@ -55,7 +55,7 @@ foreach ($path in $autorestFilesPath) {
 Write-Host "Updated autorest.md files for all the changed swaggers. `n"
 
 $packages = @()
-$dotnet = Join-Path $RepoRoot ".dotnet"
+$dotnet = Join-Path $RepoRoot "../.dotnet"
```

Comment:
How sure are we that this relative path will be the best place to install? We should potentially consider another way to get a tools working directory setup. Is there a better path that the sdk automation can give us a tools directory? We might even consider using a temp directory.

Reply:
There is an [Agent.ToolsDirectory](https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml) variable and seems to be a good candidate. Let me test with it.

Reply:
I'd like to try and get this fix it today so if this one works lets get it merged in first and then you can test with other options if you like. I think we might need to be careful depending on Agent.ToolsDirectory as well because that is very DevOps specific which will break if we do this outside of that context.

Reply:
I tried with [this](https://github.com/openapi-env-test/azure-sdk-for-net/blob/e2729ca08726a0cd5f24b1a493b64a81aaf576b4/eng/scripts/Automation-Sdk-Generate.ps1#L58-L63), but `$Env:AGENT_TOOLSDIRECTORY` doesn't seem to be set in the ADO agent as I got the same path as before or I'm accessing it in the wrong way.

Reply:
@fengzhou-msft did you update both the init and generate script? From the commit you linked to it looks like you only updated the generate. I do think it would be worth adding that environment variable as a potential override. So something like this in both the init and generate scripts:
```
$dotnet = $env:AGENT_TOOLSDIRECTORY
if (!$dotnet -or !(Test-Path $dotnet)) {
 $dotnet = Join-Path $RepoRoot .. -Resolve
}
$dotnet = Join-Path $dotnet "dotnetsdk"
```